Issues/#8454 document theme builder am (#861)
gravitee-io/gravitee-docs: pages/am/3.x/user-guide/branding/user-guide-branding-theme-builder.adoc

= Theme builder
:page-sidebar: am_3_x_sidebar
:page-permalink: am/current/am_userguide_branding_theme_builder.html
:page-folder: am/user-guide
:page-layout: am

[label label-version]#New in version 3.19#

== Overview

> Theme builder helps to create unique look and feel for the AM templates to achieve brand requirements according to your need.

Reviewer: ```suggestion
The Theme Builder enables you to create a unique look and feel for your AM templates so they are aligned with your brand requirements.
```

> Theme builder makes it easier to add custom CSS, organization's logo in the templates and preview the changes all in one place.

Reviewer: ```suggestion
The Theme Builder makes it easier to add custom CSS and the organization's logo in the templates, and to preview the changes - all in one place.
```

image::{% link images/am/current/graviteeio-am-userguide-branding-theme-builder.png %}[Theme builder,650,500]

> Theme builder is available in the domain `*Design*` section. Here is a brief summary of the builder itself:

Reviewer: ```suggestion
The Theme Builder is available in the domain `*Design*` section. A brief description of the builder follows below:
```

> * The `*Settings*` consists of the following sections:

Reviewer: ```suggestion
* The `*Settings*` include the following sections:
```

>     ** `*General*`: inputs for logo, favicon URL and theme color palette

Reviewer: ```suggestion
    ** `*General*`: inputs for logo, favicon URL, and theme color palette.
```

>     ** `*Custom CSS*`: place to write css for more advanced user

Reviewer: ```suggestion
    ** `*Custom CSS*`: a place to add custom CSS (suitable for more advanced users).
```

> * The `*Preview*` section, right-hand side of the `*Settings*` which let you switch between preview and HTML mode for more advanced customizati

Reviewer: ```suggestion
* The `*Preview*` section on the right-hand side of `*Settings*` lets you switch between preview mode and HTML mode - suitable for more advanced customizations.
```

> * `*PUBLISH*` button: saves custom theme and templates

Reviewer: ```suggestion
* `*PUBLISH*` button: saves the custom theme and templates.
```

> * `*RESET THE THEME*` button: delete the custom theme settings and reset the to the default theme

Reviewer: ```suggestion
* `*RESET THE THEME*` button: deletes the custom theme settings and resets the builder to the default theme.
```

> == Update brand logo and theme color

Reviewer: ```suggestion
== Updating the brand logo and the theme color
```

> Updating brand logo and theme color is straight forward and the changes are immediately visible in the preview section.

Reviewer: ```suggestion
It is straightforward to update the brand logo and the theme color. All changes are immediately visible in the preview section.
```

> NOTE: Logo, favicon and theme color are applied for all the default pages provided by AM.

Reviewer: ```suggestion
NOTE: Logo, favicon, and theme color are applied for all the default pages provided by AM.
```

> If you want to customize a specific page, you have to select your page and switch to the HTML mode to enable and publish custom HTML templates.

Reviewer: ```suggestion
If you want to customize a specific page, you must select your page and switch to the HTML mode to enable and publish custom HTML templates - for example, if third-party logo URLs have been used for logo and favicon and a theme color has been selected to preview the login page.
```

> As an example; 3rd party logo URLs have been used for logo and favicon and a theme color has been selected to preview the login page.

Reviewer: ```suggestion

```

> NOTE: The content Security Policy (CSP) will prevent  3rd party URL access unless it is permitted.

Reviewer: ```suggestion
NOTE: The content Security Policy (CSP) will prevent third-party URL access unless it has been set as permitted.
```

> For more information to allow cross domain URL access please check:

Reviewer: ```suggestion
For more information on how to allow cross-domain URL access, please see 
```

link:{{ '/am/current/am_installguide_how_to_configuration.html#step_7_mitigate_cross_site_scripting_xss_and_cross_site_framing' | relative_url }}[Mitigate XSS CSF in AM environment^] and

> link:{{ '/am/current//am_installguide_helm_prod_deployment.html' | relative_url }}[Mitigate XSS CSF in Helm^]

Reviewer: ```suggestion
link:{{ '/am/current//am_installguide_helm_prod_deployment.html' | relative_url }}[Mitigate XSS CSF in Helm^].
```

Author: Force pushed

image::{% link images/am/current/graviteeio-am-userguide-branding-theme-builder-custom-preview.png %}[Theme builder example,650,500]

== Custom CSS

> Customising CSS is an option for more advanced user who wants more granular custom look and feel for the html forms.

Reviewer: ```suggestion
Customising CSS is an option for more advanced users who want a more granular custom look and feel for the HTML forms.
```

Author: Force pushed

> Custom CSS should be added as shown below. Any change should be viewable in the preview section instantly.

Reviewer: ```suggestion
To add custom CSS, follow the example below. All changes will be instantly viewable in the preview section.
```

Author: Force pushed

[source,css]
----
:root {
 --variable-one-name: value;
 --variable-two-name: value;
}
----

image::{% link images/am/current/graviteeio-am-userguide-branding-theme-builder-custom-css.png %}[custom CSS,650,350]

> Check the link below for all the available CSS properties and their default values.

Reviewer: ```suggestion
Check the following documentation for information about all available CSS properties and their default values.
```

Author: Force pushed
